src/hooks/useSyncWidgetSettings.ts

```diff
@@ -144,18 +154,32 @@ export function useRecipientAddress(chainId: number) {
 }
 
 export function useWidgetFromToken() {
-  const { fromTokens } = useAtomValue(widgetSettingsAtom)
+  const { fromTokens, presetFromToken } = useAtomValue(widgetSettingsAtom)
+  const userHasSelected = useAtomValue(userSelectedFromToken)
```

Comment:
I don't quite understand the purpose of `userSelectedFromToken` and why making it undefined works.
Also is it something that can be avoided? At least not knowing too much on how the widget is implemented feels like shouldn't be necessary.

If it is necessary please do leave the comment in the code on the purpose and that should work.

Reply:
`useWidgetFromToken` and the similar "To" it's a function the widget has to return which token would be the "preselected".

Before it worked so the UI would have a token preselected when the list only contained one single token. At that point you couldn't click to change the token, since only one.

Returning `undefined` basically mean "no token should be preselected".

Now we can enable presets even when the list is longer than 1, and for that what I did was add a new atom that indicates whether the user clicked a token on the list. It starts with `false`, and goes to `true` when anything is clicked.

This way once the user has selected something, the preset goes away. Otherwise the preset value was a fixed imposition.

I will add some of this comments in the code :+1: